src/arg_spec.ml

```diff
@@ -1,19 +1,23 @@
 open! Stdune
 open Import
 
-type 'a t =
-  | A        of string
-  | As       of string list
-  | S        of 'a t list
-  | Concat   of string * 'a t list
-  | Dep      of Path.t
-  | Deps     of Path.t list
-  | Target   of Path.t
-  | Path     of Path.t
-  | Paths    of Path.t list
-  | Hidden_deps    of Path.t list
-  | Hidden_targets of Path.t list
-  | Dyn      of ('a -> Nothing.t t)
+type static = Static
```

Comment:
are the constructors necessary here?

Reply:
Probably not, but it's in line with how we do things elsewhere (see OSL for example). AFAIK this is done to make sure that the type checker can prove the two phantom types are unequal.

Reply:
Oh, I see. Neat!